src/services/AuthProvider.tsx

```diff
@@ -0,0 +1,44 @@
+import { ReactNode, createContext, useContext, useState } from "react";
+
+type Auth = {
+  isAuthenticated: boolean;
+  accessToken?: string;
+  user?: string;
+};
+
+interface IAuthContext {
+  auth: Auth;
+  setAuth: React.Dispatch<React.SetStateAction<Auth>>;
+  persist: boolean;
+  setPersist: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const initialAuthContext: IAuthContext = {
+  auth: {
+    isAuthenticated: false,
+  },
+  setAuth: () => {
+    return;
+  },
+  persist: JSON.parse(localStorage.getItem("persist")?.toString() || "true"),
```

Comment:
what are we using localstorage for?

Reply:
The persist value is going to be used as a "keep me signed in" flag when the user is signing in... 

when its set to true, the user won't have to login every time even when their session hasn't expired.
when its false, the user will be required to login after closing the page or refreshing the page

it hasn't been implemented yet in the login ui...it'll be done in the next commit